### DISCUSSION TITLE
time series forecasting

### DISCUSSION THREAD

**Discussion by ArieleZanfei** [danielegrattarola/spektral#234] [Q&A]
Hi @danielegrattarola,
I am wondering if it is possible to perform time series prediction/forecasting with GCN in spektral. 
Let's say that I have a csv or txt file that has five time series with a hourly time step that are strongly correlated (I attach to you an example of the file). I was wondering if you can help me with this task and if there is some example on how perform it. Is there some example to how perform a node regression? 

Thank you in advance!
Ariele

[generated_time_serie.TXT](https://github.com/danielegrattarola/spektral/files/6669152/generated_time_serie.TXT)

**Comment by danielegrattarola** [comment]
Hi, 

it depends on how you plan to extract graphs from your time series, e.g., a single fixed graph or a sequence of graphs over time.
Do you have any idea in mind?

**Reply by ArieleZanfei** [reply]
Hi,
Thank you for your reply! The question is not easy. The task will be predicting the next value in time of one of the time series. Imagine that you will have all this time series that are correlated, and you want to train a GCN to predict one of them. So, I am thinking that the correlated time series will represents the node of the graph, and their correlation will represent the links. So, in the case of a sequence of graphs the task will be predicting the next graphs over time. But if I just want to predict one of the time series, this will be difficult using a sequence of graphs. So maybe it will be more appropriate with a single fixed graph. Does it make sense for you?
Many thanks!

**Reply by danielegrattarola** [reply]
It does make sense! In that case, you want to have a recurrent or convolutional block to process your data along the time dimension and then a GNN block to diffuse information across the graph. 

Cheers

**Reply by ArieleZanfei** [reply]
Nice. However, I am encountering some issues data related. Is there a tutorial on how to perform a node regression starting from a csv/txt data? Because I have read almost all the examples available but they are mostly related to classification with already prepared datasets. If there is a similar example to my case, can you gently post me a link? 
Many thanks!

**Reply by danielegrattarola** [reply]
There is no example like the one you're asking for, sorry. But really, it's just a matter of setting up a time series prediction model, the GNN block can be easily integrated afterward. 

I'm happy to help you if you want to post the code that you have developed so far!

**Reply by ArieleZanfei** [reply]
Thank you. I will try to develop the code and maybe after I will contact you.
